walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,127 @@
 // limitations under the License.
 
 package com.google.sps;
-
+ 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
+  private static int FULL_DAY = 24 * 60;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    ArrayList<TimeRange> availableTimeRanges = getAvailableTimeRanges(events, request, request.getDuration());
+    return availableTimeRanges;
+
+  }
+
+  // returns what events have attendees going to the requested event
+  public ArrayList<Event> checkMatchingAttendees(Collection<Event> events, MeetingRequest request) {
+    Collection<String> requestAttendees = request.getAttendees();
+    ArrayList<Event> eventsWithMatchingAttendees = new ArrayList<Event>();
+    
+    for (String attendee : requestAttendees) {
+      for (Event event : events) {
+        ArrayList<String> eventAttendees = new ArrayList<String>(event.getAttendees());
+        if (eventAttendees.contains(attendee)) {
+          eventsWithMatchingAttendees.add(event);
+        }
+      }
+    }
+
+    return eventsWithMatchingAttendees;
+  }
+
+//   public ArrayList<TimeRange> getincovenientTimeRangesSorted(Collection<Event> events) {
+//     ArrayList<TimeRange> incovenientTimeRanges = new ArrayList<TimeRange>();
+
+//     for (Event event : events) {
+//       incovenientTimeRanges.add(event.getWhen());    
+//     }
+
+//     Collections.sort(incovenientTimeRanges, TimeRange.ORDER_BY_START);
+//     return incovenientTimeRanges;
+//   }
+
+  private ArrayList<TimeRange> getIncovenientTimeRanges(Collection<Event> events, Collection<String> requestAttendees) {
+    HashSet<String> requestAttendeesSet = new HashSet<String>();
+
+    for (String attendee : requestAttendees) {
+      requestAttendeesSet.add(attendee);
+    }
+
+    ArrayList<TimeRange> incovenientTimeRanges = new ArrayList<TimeRange>();
+    for (Event event : events) {
+      Collection<String> eventAttendees = event.getAttendees();
+      TimeRange eventTimeRange = event.getWhen();
+      for (String attendee : eventAttendees) {
+        if (eventAttendees.contains(attendee)) {
```

Comment:
It seems like this is very similar to checkMatchingAttendees? It does basically the same thing, but returns the TimeRanges of the Events instead of the Events themselves?

Is there a way to reduce code duplication? Instead we could do:
```

ArrayList<Event> matchingEvents = checkMactchingAttendees(...);
ArrayList<TimeRange> inconvenientTimeRanges;
for (Event event : matchingEvents) {
  inconvenientTimeRanges.add(event.getWhen());
}
```